tls_main.c

```diff
@@ -662,6 +694,11 @@ static int do_tls_setsockopt(struct sock *sk, int optname, char __user *optval,
 		rc = do_tls_lua_setrecv(sk, optval, optlen);
 		release_sock(sk);
 		break;
+	case TLS_LUA_WWWROOT:
```

Comment:
Why not load the `WWWROOT` value inside a Lua script? I believe doing so would really simplify your code.